resolwe/flow/managers/listener/redis_cache.py

```diff
@@ -0,0 +1,387 @@
+        if not model and (identifiers or field_name):
+            raise RuntimeError(
+                "When model is None then identifiers and field name must be provided."
+            )
+        if field_name and not identifiers:
+            raise RuntimeError(
+                "When field name is not None then identifiers must be provided."
+            )
+
+        from resolwe.test import is_testing  # Remove circular import.
+
+        parts = [
+            "listener" if not is_testing() else f"listener-{getpid()}",
+            self._model_str(model),
+            self._identifiers_str(identifiers),
+            field_name,
+        ]
+        return ":".join([part for part in parts if part])
+
+    def mget(
+        self, model: models.Model, identifiers_list: Sequence[Identifier]
+    ) -> list[FieldValues]:
+        """Obtain the set of fields from the redis cache.
+
+        All the cached field names are always returned.
+
+        The query does not run in a transaction.
+        """
+        redis_keys = [
+            self.get_redis_key(model, identifier) for identifier in identifiers_list
+        ]
+        cached_data = list()
+
+        for counter in range(0, len(redis_keys), self.KEY_BATCH_SIZE):
+            batch_keys = redis_keys[counter : counter + self.KEY_BATCH_SIZE]
+            try:
+                cached_data.extend(self._redis.mget(*batch_keys))  # type: ignore
+            except redis.exceptions.RedisError:
+                logger.exception(
+                    __(
+                        "Could not retrieve data from Redis for keys: {}",
+                        ", ".join(redis_keys),
+                    )
+                )
+                raise
+
+        to_return = list()
+        for counter, data in enumerate(cached_data):
+            to_append = None
+            if data is not None:
+                try:
+                    to_append = pickle.loads(data)
+                except pickle.PickleError:
+                    redis_key = redis_keys[counter]
+                    logger.exception(
+                        __(
+                            "Could not deserialize data from Redis for key '{}'",
+                            redis_key,
+                        )
+                    )
+            to_return.append(to_append)
+        return to_return  # type: ignore
+
+    def get(
+        self,
+        model: models.Model,
+        identifiers: Identifier,
+        field_names: Iterable[str],
+    ) -> FieldValues:
+        """Obtain the set of fields from the redis cache.
+
+        If field_name is requested that is not cached the rusult for that key is None.
+
+        The query is run in a transaction.
+        """
+
+        def get_redis_data(
+            pipeline: redis.client.Pipeline,
+        ) -> dict[str, Optional[Union[str, datetime]]]:
+            """Get the redis data from the specific key."""
+
+            cached_data = dict()
+            with suppress(Exception):
```

Comment:
Why do you handle all possible errors in the `mget` method, and just suppress them here?